src/System.Linq.Expressions/src/System/Linq/Expressions/Compiler/LambdaCompiler.Expressions.cs

```diff
@@ -862,24 +860,6 @@ private void EmitMemberGet(MemberInfo member, Type objectType)
             }
         }
 
-        private static bool TryGetRawConstantValue(FieldInfo fi, out object value)
-        {
-            // TODO: It looks like GetRawConstantValue is not available at the moment, use it when it comes back.
-            //value = fi.GetRawConstantValue();
-            //return true;
-
-            try
-            {
-                value = fi.GetValue(obj: null);
-                return true;
-            }
-            catch
-            {
-                value = null;
```

Comment:
I am not sure what the try/catch was trying to achieve here. It is illegal to reference literal field in IL so fallback to "EmitFieldGet" would be pointless. 

Perhaps the TryGetRawConstantValue was written before the caller started checking for  IsLiteral...